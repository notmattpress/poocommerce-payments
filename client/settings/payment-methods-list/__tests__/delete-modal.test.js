/** @format */

/**
 * External dependencies
 */
import React, { act } from 'react';
import { render, screen } from '@testing-library/react';
import user from '@testing-library/user-event';

// Mock ResizeObserver for modal components - comprehensive setup
class MockResizeObserver {
	observe() {}
	unobserve() {}
	disconnect() {}
}

// Set up ResizeObserver mock in multiple places to ensure it's available
global.ResizeObserver = MockResizeObserver;
global.window = global.window || {};
global.window.ResizeObserver = MockResizeObserver;

// Ensure it's available on window in JSDOM environment
Object.defineProperty( window, 'ResizeObserver', {
	writable: true,
	configurable: true,
	value: MockResizeObserver,
} );

/**
 * Internal dependencies
 */
import ConfirmPaymentMethodDeleteModal from '../delete-modal';
import paymentMethodsMap from 'wcpay/payment-methods-map';

const mockOnClose = jest.fn();
const mockOnConfirm = jest.fn();

describe( 'Activation Modal', () => {
	beforeAll( () => {
		global.wcpaySettings = {
			accountEmail: 'admin@example.com',
		};
	} );
	afterEach( () => {
		jest.resetAllMocks();
	} );

	const renderDeleteModal = () => {
		return render(
			<ConfirmPaymentMethodDeleteModal
				id={ 'card' }
				label={ paymentMethodsMap.card.label }
				icon={ paymentMethodsMap.card.icon }
				onCancel={ mockOnClose }
				onConfirm={ mockOnConfirm }
			/>
		);
	};

	it( 'matches the snapshot', () => {
		const container = renderDeleteModal().baseElement;
		expect( container ).toMatchSnapshot();
	} );

	it( 'contains the label of the payment method', () => {
		renderDeleteModal();
		expect(
			screen.queryByText( paymentMethodsMap.card.label )
		).toBeInTheDocument();
	} );

	it( 'triggers the onClose event on close button click', async () => {
		renderDeleteModal();
		const closeButton = screen.queryByRole( 'button', { name: 'Cancel' } );
		expect( mockOnClose ).not.toBeCalled();
		await act( async () => {
			await user.click( closeButton );
		} );
		expect( mockOnClose ).toHaveBeenCalled();
	} );

	it( 'triggers the onConfirmClose event on confirm button click', async () => {
		renderDeleteModal();
		const confirmButton = screen.queryByRole( 'button', {
			name: 'Remove',
		} );
		expect( mockOnConfirm ).not.toHaveBeenCalled();
		await act( async () => {
			await user.click( confirmButton );
		} );
		expect( mockOnConfirm ).toHaveBeenCalled();
	} );
} );
