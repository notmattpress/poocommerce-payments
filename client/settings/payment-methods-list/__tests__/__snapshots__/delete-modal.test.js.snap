// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Activation Modal matches the snapshot 1`] = `
.emotion-0 {
  margin-bottom: 0;
  margin-left: calc(4px * 2);
}

<body
  class="modal-open"
>
  <p
    class="a11y-speak-intro-text"
    hidden="hidden"
    id="a11y-speak-intro-text"
    style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
  >
    Notifications
  </p>
  <div
    aria-atomic="true"
    aria-live="assertive"
    aria-relevant="additions text"
    class="a11y-speak-region"
    id="a11y-speak-assertive"
    style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
  />
  <div
    aria-atomic="true"
    aria-live="polite"
    aria-relevant="additions text"
    class="a11y-speak-region"
    id="a11y-speak-polite"
    style="position: absolute;margin: -1px;padding: 0;height: 1px;width: 1px;overflow: hidden;clip: rect(1px, 1px, 1px, 1px);-webkit-clip-path: inset(50%);clip-path: inset(50%);border: 0;word-wrap: normal !important;"
  />
  <div
    aria-hidden="true"
  />
  <div
    class="components-modal__screen-overlay"
  >
    <div
      aria-labelledby="components-modal-header-0"
      class="components-modal__frame wcpay-confirmation-modal"
      role="dialog"
      style="--modal-frame-animation-duration: 200ms;"
      tabindex="-1"
    >
      <div
        class="components-modal__content"
        role="document"
      >
        <div
          class="components-modal__header"
        >
          <div
            class="components-modal__header-heading-container"
          >
            <h1
              class="components-modal__header-heading"
              id="components-modal-header-0"
            >
              Remove Credit / Debit Cards from checkout
            </h1>
          </div>
          <div
            class="components-spacer emotion-0 emotion-1"
            data-wp-c16t="true"
            data-wp-component="Spacer"
          />
          <button
            aria-label="Close"
            class="components-button is-compact has-icon"
            type="button"
          >
            <svg
              aria-hidden="true"
              focusable="false"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m13.06 12 6.47-6.47-1.06-1.06L12 10.94 5.53 4.47 4.47 5.53 10.94 12l-6.47 6.47 1.06 1.06L12 13.06l6.47 6.47 1.06-1.06L13.06 12Z"
              />
            </svg>
          </button>
        </div>
        <div>
          <div
            class="payment-delete-illustration__wrapper"
          >
            <div
              class="payment-delete-illustration__illustrations"
            >
              <img
                alt="Credit / Debit Cards"
                class="payment-method__icon payment-delete-illustration__payment-icon"
                src="assets/images/icon-card.svg"
              />
              <svg
                class="gridicon gridicons-cross-circle payment-delete-illustration__payment-cross-icon"
                height="24"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g>
                  <path
                    d="M19.1 4.9C15.2 1 8.8 1 4.9 4.9S1 15.2 4.9 19.1s10.2 3.9 14.1 0 4-10.3.1-14.2zm-4.3 11.3L12 13.4l-2.8 2.8-1.4-1.4 2.8-2.8-2.8-2.8 1.4-1.4 2.8 2.8 2.8-2.8 1.4 1.4-2.8 2.8 2.8 2.8-1.4 1.4z"
                  />
                </g>
              </svg>
            </div>
          </div>
          <p>
            Are you sure you want to remove 
            <strong>
              Credit / Debit Cards
            </strong>
            ? Your customers will no longer be able to pay using Credit / Debit Cards.
          </p>
          <p>
            You can add it again at any time in 
            <a
              href="admin.php?page=wc-settings&tab=checkout&section=poocommerce_payments"
            >
              WooPayments
            </a>
            .
          </p>
          <hr
            class="wcpay-confirmation-modal__separator"
          />
          <div
            class="wcpay-confirmation-modal__footer"
          >
            <button
              class="components-button is-primary is-destructive"
              type="button"
            >
              Remove
            </button>
            <button
              class="components-button is-secondary"
              type="button"
            >
              Cancel
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
